mlfinlab/portfolio_optimization/cla.py

```diff
@@ -0,0 +1,608 @@
+'''
+This module implements the famous Critical Line Algorithm for mean-variance portfolio
+optimisation. It is reproduced with modification from the following paper:
+D.H. Bailey and M.L. Prado “An Open-Source Implementation of the Critical- Line Algorithm for
+Portfolio Optimization”,Algorithms, 6 (2013), 169-196. http://dx.doi.org/10.3390/a6010169
+'''
+
+import numbers
+from math import log, ceil
+import numpy as np
+import pandas as pd
+
+
+def _infnone(number):
```

Comment:
Why is this logic outside the class? Can we put it in the class, if it is the only place it is used?

Reply:
@Jackal08 I already put it inside the class but havent commited. Will be reflected in the next commit